src/index.js

```diff
@@ -216,16 +216,13 @@ class JanusAdapter {
       handle.sendTrickle(event.candidate);
     });
 
-    var offer = await peerConnection.createOffer({
-      offerToReceiveAudio: true
-    });
-
-    await peerConnection.setLocalDescription(offer);
-    var answer = await handle.sendJsep(offer);
-    await peerConnection.setRemoteDescription(answer.jsep);
-
     // Send join message to janus. Don't listen for join/leave messages. Subscribe to the occupant's audio stream.
-    await this.sendJoin(handle, this.room, this.userId, { notifications: false, media: occupantId });
+    const resp = await this.sendJoin(handle, this.room, this.userId, { notifications: false, media: occupantId });
+
+    await peerConnection.setRemoteDescription(resp.jsep);
+    const answer = await peerConnection.createAnswer();
+    await peerConnection.setLocalDescription(answer);
+    await handle.sendJsep(peerConnection.localDescription);
```

Comment:
This looks correct AFAIK but you should be parallelizing steps instead of doing it all serially.

(The rest of this code in general looks buggy, because you're not waiting for ICE to finish or for `webrtcup` to fire before you get the streams. I think it only works by luck. Someone should probably think for half an hour and try to make it really correct.)

Reply:
Yeah I agree with Marshall, I think you need to wait for `webrtcup` before getting the streams. There's a potential race condition there. I don't think that anything should be parallelized though. This looks pretty good.

Reply:
You don't think you could do this?
```
await peerConnection.setRemoteDescription(resp.jsep);
var answerReady = peerConnection.createAnswer();
var localReady = answerReady.then(a => peerConnection.setLocalDescription(a));
var remoteReady = answerReady.then(a => handle.sendJsep(a));
await Promise.all([localReady, remoteReady]);
```
To be clear, I'm not 100% sure that's correct, but it seems right.

Reply:
I don't see anything in the [spec](https://w3c.github.io/webrtc-pc/#dom-rtcpeerconnection-setlocaldescription) that says you shouldn't send the answer jsep before setting the local description. So I suppose you are able to parallelize those two things and shave some milliseconds off the connection time.

Reply:
:+1: I had to add a mechanism to wait for the right `webrtcup` event. Please take a look.